main.tf

```diff
@@ -108,6 +108,33 @@ resource "aws_lambda_function" "this" {
   depends_on = [null_resource.archive, aws_s3_object.lambda_package, aws_cloudwatch_log_group.lambda]
 }
 
+# This resource contains the extra information required by SAM CLI to provide the testing capabilities
+# to the TF application. The required data is where SAM CLI can find the Lambda function source code
+# and what are the resources that contain the building logic.
+resource "null_resource" "sam_metadata_aws_lambda_function_this" {
+  count = local.create && var.create_package && var.create_function && !var.create_layer ? 1 : 0
+  triggers = {
+    # This is a way to let SAM CLI correlates between the Lambda function resource, and this metadata
+    # resource
+    resource_name = "aws_lambda_function.this[0]",
+    resource_type = "ZIP_LAMBDA_FUNCTION",
+
+    # The Lambda function source code.
+    original_source_code = jsonencode(var.source_path),
```

Comment:
source_path can be rather complicated type (see examples or readme), I am not sure this will be well supported by SAM. Could you please verify this?

Reply:
In Sam Cli, we expect the value of this property to be either String or JSON object. In case if it is a JSON object as we have here in serverless TF module, SAM CLI will expect another property `source_code_property` to let SAM CLI know, where it can find the Lambda resource code path. In Sam Cli , we do not support multiple paths, so in case if the value of this attribute is a list (of strings, or objects), SAM CLI will process only the first item in the list.